src/plugins/vis_type_timeseries/public/application/components/vis_editor.js

```diff
@@ -36,6 +36,12 @@ import { Storage } from '../../../../../plugins/kibana_utils/public';
 const VIS_STATE_DEBOUNCE_DELAY = 200;
 const APP_NAME = 'VisEditor';
 
+const debouncedFetchFields = debounce(
+  (extractedIndexPatterns) => fetchFields(extractedIndexPatterns),
+  VIS_STATE_DEBOUNCE_DELAY,
+  { leading: true }
```

Comment:
is it required?

Reply:
Because `fetchFields` is a debounced function, we can't access it result immediately in `then`:
`this.fetchFields(extractedIndexPatterns).then((visFields) =>`

Without the leading option, in the console will appear error like this:
![image](https://user-images.githubusercontent.com/54894989/100452786-e0531380-30ca-11eb-96c7-9d86f4548bf9.png)

So `{ leading: true }`  is used here to invoke `fetchFields` as soon as typing started to access the fields